R/consecutive_mutate_linter.R

```diff
@@ -0,0 +1,76 @@
+#' Force consecutive calls to mutate() into just one when possible
+#'
+#' `dplyr::mutate()` accepts any number of columns, so sequences like
+#'   `DF %>% dplyr::mutate(..1) %>% dplyr::mutate(..2)` are redundant --
+#'   they can always be expressed with a single call to `dplyr::mutate()`.
+#'
+#' An exception is for some SQL back-ends, where the translation logic may not be
+#'   as sophisticated as that in the default `dplyr`, for example in
+#'   `DF %>% mutate(a = a + 1) %>% mutate(b = a - 2)`.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "x %>% mutate(a = 1) %>% mutate(b = 2)",
+#'   linters = consecutive_mutate_linter()
+#' )
+#'
+#' # okay
+#' code <- "library(dplyr)\nx %>% mutate(a = 1) %>% mutate(a = a + 1)"
+#' writeLines(code)
+#' lint(
+#'   text = code,
+#'   linters = consecutive_mutate_linter()
+#' )
+#'
+#' @evalRd rd_tags("consecutive_mutate_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+consecutive_mutate_linter <- function(invalid_backends = "dbplyr") {
+  blocked_library_xpath <- glue("
```

Comment:
That seems quite brittle. What about ns-qualified usages of dbplyr? What about `require()`?

Reply:
Very brittle. Especially in packages... I don't have a great general solution here

> `require()`

We already migrated all code to use `library()` so omitted that. I'll add here.

> ns-qualified usages of dbplyr

SGTM. I'll also look for `#' @import` comments (which later can be improved by using proper roxygen2 parsing)